### PR TITLE
Razee multicluster impl no tests

### PR DESCRIPTION
**Pull request by tomersolomon1**
Razee cluster-manager implementation. This PR doesn't include tests. Tests require pushing stuff to different cluster, and will be handled in future PR.

### CODE REVIEW

File: pkg/multicluster/razee/razee.go [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+		return "", err
+	}
+	return result.AddGroup.Uuid, nil
+}
+
+//nolint:golint,unused
+type IAMRazeeClient struct {
+	apiKey string
+	client *graphql.Client
+}
```
Comment (roee88) [inline pkg/multicluster/razee/razee.go:189]: Not needed

File: pkg/multicluster/razee/razee_cluster_manager.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+	return nil
+}
+
+//nolint:golint,unused
+func channelName(cluster string, namespace string, name string) string {
+	return "m4d-" + cluster + "-blueprint-" + namespace + "-" + name
+}
+
+//nolint:golint,unused
+func NewRazeeManager(url string, login string, password string, orgId string) multicluster.ClusterManager {
```
Comment (roee88) [inline pkg/multicluster/razee/razee_cluster_manager.go:222]: Shouldn't this accept a RazeeClient instead?

File: pkg/multicluster/razee/razee_cluster_manager.go [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+	}
+	if removeChannel.Success {
+		r.log.Info("Successfully deleted channel " + removeChannel.UUID)
+	}
+	return nil
+}
+
+//nolint:golint,unused
+func channelName(cluster string, namespace string, name string) string {
+	return "m4d-" + cluster + "-blueprint-" + namespace + "-" + name
```
Comment (roee88) [inline pkg/multicluster/razee/razee_cluster_manager.go:218]: can we swap to `m4d.ibm.com/<cluster>/<namespace>/<name>`? or is it illegal?

**Review by roee88 — COMMENTED** [review]
Thanks, looks good. @tomersolomon1 I did a first pass review so I will review the actual logic tomorrow and let @froesef know in case of any suspected issue.

File: pkg/multicluster/razee/razee_cluster_manager.go (hunk 2 above)
Comment (tomersolomon1) [reply pkg/multicluster/razee/razee_cluster_manager.go:222]: Not sure I understand. Maybe you mean Round-Tripper?

File: pkg/multicluster/razee/razee_cluster_manager.go (hunk 3 above)
Comment (tomersolomon1) [reply pkg/multicluster/razee/razee_cluster_manager.go:218]: Nothing is illegal in wonderland:
```
> db.channels.find()
{ "_id" : "a3fe3acb-155f-4fc2-90da-ba17fbb90855", "uuid" : "15b09186-5103-4fbe-bddd-e9ad785e7ebe", "org_id" : "e9077db1-f82f-4cdf-a0e6-06dc7528e0f6", "name" : "m4d.ibm.com/cluster-1/namespace-2/resource-3", "versions" : [ ], "created" : ISODate("2020-12-07T16:20:48.410Z"), "__v" : 0 }
> 
```

File: pkg/multicluster/razee/razee.go (hunk 1 above)
Comment (tomersolomon1) [reply pkg/multicluster/razee/razee.go:189]: Right

File: pkg/multicluster/razee/razee_cluster_manager.go (hunk 2 above)
Comment (froesef) [reply pkg/multicluster/razee/razee_cluster_manager.go:222]: We have two different RazeeClients at the moment. Both of them have a different initialization. I would hide the Razee specific stuff behind these functions and not expose it further up.

Comment (roee88) [reply pkg/multicluster/razee/razee_cluster_manager.go:222]: What two clients? All of the New*Client functions return RazeeClient. 

My suggestion was that NewRazeeManager  accept a RazeeClient. Usage is for example:
`NewRazeeClient(NewGithubAPIKeyClient(...))` but you will be able to initiate it with Local or IAM the same way.

Comment (froesef) [reply pkg/multicluster/razee/razee_cluster_manager.go:222]: `con` and `razeeClient`. We currently use multiple clients to razee as the library does not support all functions yet. (The plan is to eventually implement everything in sat-con library and then have only that one)

**Review by froesef — APPROVED** [review]

File: pkg/multicluster/razee/razee_cluster_manager.go (hunk 2 above)
Comment (roee88) [reply pkg/multicluster/razee/razee_cluster_manager.go:222]: Does `con` only supports local authentication?

**Review by roee88 — APPROVED** [review]
Approving to avoid blocking further work, but let's have a sync tomorrow because I do think this needs some quick changes in a follow-up.

**Pull request merged**